lib/peep/buckets/custom.ex

```diff
@@ -6,140 +6,85 @@ defmodule Peep.Buckets.Custom do
   For an example, look at the source of `Peep.Buckets.PowersOfTen`.
   """
 
-  defmacro __using__(opts) do
-    buckets =
-      Keyword.fetch!(opts, :buckets)
-      |> :lists.usort()
-
-    unless Enum.all?(buckets, &is_number/1) do
-      raise ArgumentError, "expected buckets to be a list of numbers, got: #{buckets}"
-    end
+  defmacro __using__(buckets: buckets) do
+    quote do
+      @buckets :lists.usort(unquote(buckets))
 
-    number_of_buckets = length(buckets)
+      unless Enum.all?(@buckets, &is_number/1) do
+        raise ArgumentError, "expected buckets to be a list of numbers, got: #{@buckets}"
+      end
 
-    quote do
       @behaviour Peep.Buckets
 
       @impl true
       def config(_), do: %{}
 
       @impl true
-      def number_of_buckets(_), do: unquote(number_of_buckets)
+      def number_of_buckets(_), do: length(@buckets)
```

Comment:
Can compute this comp time as well (not sure if the compiler notices it's a constant expr)